libraries/Microsoft.Bot.Connector.Streaming/Transport/NamedPipeTransport.cs

```diff
@@ -152,6 +134,97 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        private async Task RetryToSucceedAsync(Func<CancellationToken, Task> createNamedPipeFactory, CancellationToken cancellationToken)
+        {
+            _tryTimes = 0;
+            while (!cancellationToken.IsCancellationRequested)
```

Comment:
You can be more concise by saying 

while (!cancellationToken.IsCancellationRequested && _tryTimes++ < 100) ?